spec/configuration_spec.lua

```diff
@@ -125,6 +125,45 @@ describe('Configuration object', function()
       assert.same(services, filter_services(services, { '42' }))
       assert.same({}, filter_services(services, { '21' }))
     end)
+
+    describe("with service filter", function()
+      local mockservices = {
+        {id="42", hosts={"test.foo.com"}},
+        {id="12", hosts={"staging.foo.com"}},
+        {id="21", hosts={"prod.foo.com"}},
+      }
+
+      it("validates default case", function()
+        assert.same(filter_services(mockservices, nil), mockservices)
+      end)
+
+      it("reads from environment variable", function()
```

Comment:
I think it would also be good to test a regex that matches more than one but not all the services. Just to make sure that the code does not stop at the first match.